### ISSUE TITLE
excel  becomes laggy and unresponsive under certain circumstances.

### ISSUE THREAD

**Issue by stickbear2015** [nvaccess/nvda#9595]
### Steps to reproduce:
Open the attached .xls file in excel
### Actual behavior:
NVDA initially works but as you enter data, and minnipulate this form NVDA lags and becomes unresponsive while in the excel window. While this doesn't prevent working in excel it slows things down.
### Expected behavior:
NVDA should be just as responsive in excel as in other parts of the operating system.
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
2019.1.1
#### Windows version:
Microsoft Windows
Version 1809 (OS Build 17763.437) 
#### Name and version of other software in use when reproducing the issue:
Microsoft Excel Subscription Version 16.0.11601.20204
#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
same issue persists all the way back through NVDA 2018.1.
[ohio, (ood), state report template blank.xlsx](https://github.com/nvaccess/nvda/files/3196117/ohio.ood.state.report.template.blank.xlsx)

**Comment by josephsl** [comment]
Hi,

Can you try this with latest alpha and beta snapshots please?

Thanks.

**Comment by stickbear2015** [comment]
@josephsl 
Testing with latest alpha and beta snapshots results in similar performance issues.
I also for fun ran latest master bleeding edge with similar issues.

**Comment by josephsl** [comment]
Hi,

Master = alpha.

You mentioned this not happening with really old releases. Can you pinpoint to a specific version (say, 2017.4) where NVDA responded just as quickly as any other scenarios? If a later version did introduce regressions, chances are that it might be something with NVDA, Excel, or a combination of these.

Thanks.

**Comment by stickbear2015** [comment]
@josephsl 
I've run tests so far back to when we began supporting 2016. This particular form is full of formulas, and other administrative nonsense I don't even know if I can find a version that'll be as responsive as you'd expect in office. Have you tried playing with the file I attached?

**Comment by josephsl** [comment]
Hi, yes I have, and didn’t notice differences yet. Thanks.

**Comment by Qchristensen** [comment]
It doesn't seem too sluggish for me either.  Every single field has an input message which pops up, I wonder if that could be causing it for you?

Personally, I wouldn't have designed it with an input message on EVERY field.  EG instead of having the label "Invoice date" and the input message clarifying to include the date in MM/DD/YYYY, just make the label "Invoice date (MM/DD/YYYY)".

So, my two questions then are:
1) Does it ONLY happen with this spreadsheet?  If you load other spreadsheets, is it also sluggish?
2) Could you provide a log (ideally at debug level) - you can set the log level in NVDA's general options NVDA+control+g, and you can find the log in your %temp% folder, or press NVDA+F1 after you've reproduced the sluggishness and copy the text (either here or in an email to info@nvaccess.org and mention this issue so I know what it relates to).

**Comment by Adriani90** [comment]
Which Synthesizer are you using? Do you have addons installed? Could you provide a list of nvda addons installed?
In Excel and in MS Office in General, you can also install addins. Could you disabel the addins in Excel and see if it is still sluggisch? Sometimes the dropbox addin causes Performance issues with NVDA.

**Comment by stickbear2015** [comment]
go ahead and try filling out the form, then leave and come back to it. Just fill it with example data. note: it will come up protected, enable editing to fill it out.
I'm using codefactories eliquence addon, even with others disabled it still gets sluggish as data is entered. It's more noticeable if I leave the window and come back to it to continue working in the file, say for example to look up a clients information or to review notes from a client visit.

**Comment by stickbear2015** [comment]
I'm attaching the debug output from NVDA 2019.2 beta 1 to this comment. I don't know what's gonna be useful, but just before I entered the log viewer things were starting to get slow and sluggish, so I hope their's something of use here
[nvda-debug-2019-beta2.log](https://github.com/nvaccess/nvda/files/3265368/nvda-debug-2019-beta2.log)
.

**Comment by stickbear2015** [comment]
I'm prodding this issue again as I haven't seen any updates on it since I posted the log.

**Comment by ivnc** [comment]
Hi.

A spanish-speaking user has reported us through Twitter to have a similar problem with any Excel spreadsheet.
When entering a formula or even some (random) text, Excel becomes "very slow" since an apparently random point, forcing the user to restart the application. In some cases, he has needed to restart his PC in order to temporarily restore the normal behavior.
Disabling the "Allow editing directly on cells" option mitigates the problem, but the trick was done by disabling "Report with sounds" from Options>Accessibility. With that option both enabled or disabled I'm unable to reproduce the problem in this computer, but the user says that this problem apears while sounds are turned on and disappears with them turned off.
Could this be the same issue? Specially for @stickbear2015. I suggest to try if disabling sounds serves as a patch for the issue. If not, creating an independent ticket might be the best option.
I attach a log submitted by the user where he reproduces the problem and the details about our testing environments.
* [User's log file (debug)](https://github.com/nvaccess/nvda/files/3360598/core_9595-JMecco-relevant_part.log) The line starting with "#" is a comment that replaces an ommitted fragment.
* Mine: NVDA 2019.2 beta-17939 with Office 2019 (16.0.11727.20222 64 bit) on windows 10 1803 (build 17134.829).
* His: NVDA 2019.2 beta1 with Office365 (16.0.11629.20238 32 bit) on Windows. He didn't try to reproduce with any previous versions.

Regards.

**Comment by Adriani90** [comment]
@stickbear2015 it seems there is some code laoded from global plugins from your user config folder in Dropbox. Did you copy there any code? This seems to cause the errors. There is also something weird with the user configuration. Please uninstall NVDA complemtely, after that open narator, press windows+r and type %appdata%
Then go to nvda and delete the whole folder with shift+delete.
After that install NVDA again and see if the issue still occurs.
Don't forget to report back on this issue otherwise people will think that this is abandoned.
Thanks.

**Comment by Adriani90** [comment]
We do not have an update since almost 10 months, closing as abandoned. Please comment on it if it is still occuring with the last versions and we can reopen it. Thanks.

**Comment by Neurrone** [comment]
I don't think we should close the issue so hastily. NVDA's issues with office, excel included are well known so I think we should still leave this open unless the STR no longer reproduces the problem.

**Comment by Adriani90** [comment]
It is not hastly, we asked for updates and didn't get any answer since months. I also provided suggestions in my comment. If the initial author reports that this is still reproducible, we can reopen.
I cannot reproduce it with last NVDA alpha on Excel 2016.
